errors/httperrors/errors.go

```diff
@@ -36,3 +38,26 @@ func IsResponseStatusError(err error) bool {
 	_, isRespErr := err.(*ResponseStatusError)
 	return isRespErr
 }
+
+func GetStatusCodeFromResponseStatusError(err error) (int, error) {
```

Comment:
I suggest removing the GetStatusCodeFromResponseStatusError function, because its usage is a bit dangerous.
As a replacement for this function I suggest the following:
In all the places this function is current use, check the response status returned from the http client.
Do acheieve this, you may need to add the response as an additional returned parameter from the http client methods.

Reply:
Done